src/log-store/src/kafka/log_store.rs

```diff
@@ -101,24 +101,30 @@ impl LogStore for KafkaLogStore {
         }
 
         // Builds a record from entries belong to a region and produces them to kafka server.
-        let region_ids = producers.keys().cloned().collect::<Vec<_>>();
-
-        let tasks = producers
-            .into_values()
-            .map(|producer| producer.produce(&self.client_manager))
-            .collect::<Vec<_>>();
+        let (region_ids, tasks): (Vec<_>, Vec<_>) = producers
+            .into_iter()
+            .map(|(id, producer)| (id, producer.produce(&self.client_manager)))
+            .unzip();
         // Each produce operation returns a kafka offset of the produced record.
         // The offsets are then converted to entry ids.
         let entry_ids = futures::future::try_join_all(tasks)
             .await?
             .into_iter()
             .map(TryInto::try_into)
             .collect::<Result<Vec<_>>>()?;
-        debug!("The entries are appended at offsets {:?}", entry_ids);
+        let last_entry_ids = region_ids
+            .into_iter()
+            .zip(entry_ids)
+            .collect::<HashMap<_, _>>();
 
-        Ok(AppendBatchResponse {
-            last_entry_ids: region_ids.into_iter().zip(entry_ids).collect(),
-        })
+        #[cfg(debug)]
+        {
+            for (region_id, offset) in last_entry_ids.iter() {
+                debug!("Entries for region {region_id} are appended at the start offset {offset}");
```

Comment:
Maybe we need to be specific: is this log only valid in debug builds, or in debug level in whatever build?

If this log is only valid in debug builds, then:
```rs
#[cfg(debug_assertion)]
{
    for ... {
        info!(...)
    }
}
```
would be ok.

If the log is expected to print in debug level, then it should be:
```rs
    debug!("Append batch result: {:?}", last_entry_ids);
```